src/ln/peer_handler.rs

```diff
@@ -710,12 +711,31 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 						}
 						continue;
 					},
+					Event::SendErrorMessage { ref node_id, ref msg } => {
+						let (mut descriptor, peer) = get_peer_for_forwarding!(node_id, {
+								//TODO: Do whatever we're gonna do for handling dropped messages
+							});
+						peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(msg, 17)));
+						Self::do_attempt_write_data(&mut descriptor, peer);
+						continue;
+					},
+					Event::DisconnectPeer { ref node_id } => {
+						let (mut descriptor, peer) = get_peer_for_forwarding!(node_id, {
+								//TODO: Do whatever we're gonna do for handling dropped messages
+							});
+						disconnect_peers.push(descriptor.clone());
+						continue;
+					}
 				}
 
 				upstream_events.push(event);
 			}
 		}
 
+		for descriptor in disconnect_peers {
+			self.disconnect_event(&descriptor);
```

Comment:
You also need some fn in the SocketDescriptor trait that allows us to notify the network provider that we want to disconnect the peer. For simplicity of this patch, I think its fine to indicate that if its called, disconnect_event must not be called, nor any future read_events.